app/controllers/member/messages_controller.rb

```diff
@@ -2,29 +2,13 @@ class Member::MessagesController < ApplicationController
   before_action :authenticate_member!
 
   def create 
-    @message = Message.new(message_params)
-    @message.member_id = current_member.id 
+    @message = Message.message_create!(current_member, message_params)
     @room = @message.room
-    if @message.save
-      
-      @roommembernotme = Entry.where(room_id: @room.id).where.not(member_id: current_member.id)
-      @theid = @roommembernotme.find_by(room_id: @room.id)
-      notification = current_member.active_notifications.new(
-          room_id: @room.id,
-          message_id: @message.id,
-          visited_id: @theid.member_id,
-          visiter_id: current_member.id,
-          action: 'dm'
-      )
-      # 自分の投稿に対するコメントの場合は、通知済みとする
-      if notification.visiter_id == notification.visited_id
-          notification.checked = true
-      end
-      notification.save if notification.valid?
-      redirect_back(fallback_location: root_path)
-    else 
-      redirect_back(fallback_location: root_path)
-    end
+    @entry = Entry.where.not(member_id: current_member.id).find_by(room_id: @room.id)
+    current_member.active_notifications.notification_create!(current_member, @room, @message, @entry)
+    redirect_back(fallback_location: root_path)
+  rescue => e
+    redirect_back(fallback_location: root_path)
```

Comment:
`rescue` する前と後の処理で必ず行いたい処理があるのであれば `ensure` が使えます

```suggestion
rescue => e
    # do something
ensure
    redirect_back(fallback_location: root_path)
```

Reply:
cca359c 
修正しました。